0002/README.md

```diff
@@ -0,0 +1,257 @@
+# Proposal 2: Project roadmap, May 2022
+
+*   Proposer: [Mark Lodato][@MarkLodato] (lodato@google.com)
+*   GitHub Issue: n/a
+*   Status: [DRAFT](../README.md#meaning-of-status-codes)
+*   Implementation: TODO
+
+## Abstract
+
+This document proposes long-term themes and short-term milestones for the SLSA
+project.
+
+> NOTE TO READER: If you know of committed projects that should be added to this
+> roadmap, please send a pull request or file an issue in this repo.
+
+## Implementation
+
+Once this proposal is accepted, we will track progress through GitHub Projects
+or Milestones.
+
+## Themes (long-term)
+
+The following open-ended themes encapsulate the main objectives of SLSA. They
+are expected to remain stable long-term.
+
+-   **Specification** is stable, practical, and useful for reducing risk—with a
+    healthy surrounding community.
+-   **Tooling**, services, and documentation make SLSA readily adoptable.
+-   **Adoption** of SLSA by open-source projects reduces security risk for
+    users.
+-   **Government and industry** widely accept SLSA as the *lingua franca* of
+    supply chain security.
+
+## Milestones (short-term)
+
+The following milestones are expected to be practical in the next few quarters.
+Each item or sub-item has a priority and a main point of contact.
+
+Priorities are to be interpreted as follows:
+
+-   P0: High confidence it will happen mid-2022
+-   P1: Expected mid-2022 but may slip
+-   P2: Planned for 2022 but may slip or get de-prioritized
+-   P3: Side project, moderate risk of de-prioritization
+
+### Theme: Specification and community
+
+#### Milestone: SLSA specification has a stable 1.0 release ([@MarkLodato], P1)
+
+Version 1.0 of the [SLSA specification](https://slsa.dev/spec) provides a stable
+foundation. Plans:
+
+-   Pare SLSA v1.0 down to only cover build integrity, which is SLSA's core
+    strength. This makes SLSA practical by removing the requirements that cannot
+    be implemented today, notably the controversial two-party review requirement
+    and the undefined "common" requirements.
```

Comment:
Is this proposing to remove the two-party review from SLSA level 4 + others?

Reply:
Yes, that is what I plan to propose, which will need to be approved independently of this roadmap. We'll have to see if the new proposed ladder makes sense overall. Two-party review would instead become an "extension" under the next bullet, possibly as part of a larger effort to ensure the code is "good". This might also mean adding other alternatives to achieve the same goal.

Today, mandatory two-party review is impractical for most organizations, particularly open-source projects:
- Most organizations lack the culture, tooling, and infrastructure to maximize review effectiveness, minimize overhead, and ensure strong security guarantees.
- We don't yet have technical means to verify two-party review happened on any popular code hosting sites, such as GitHub or GitLab, and we don't want to block the build integrity piece of SLSA on this.

I plan to write this all up as part of the proposal. 

Are you suggesting any changes to the text, or just asking for background?

Reply:
> We don't yet have technical means to verify two-party review happened on any popular code hosting sites, such as GitHub or GitLab, and we don't want to block the build integrity piece of SLSA on this.

@MarkLodato I don't know about GitHub, but FWIW GitLab actually has several ways of verifying that two-party review happened.  First, MR approval rules can be configured to [require two-party review](https://docs.gitlab.com/ee/user/project/merge_requests/approvals/rules.html) with an option to [disallow authors and committers from approving](https://docs.gitlab.com/ee/user/project/merge_requests/approvals/settings.html).  Second, whenever review happens, it is recorded in the Audit Log and can be verified in that way.  For a more convenient and visual verification method, a [Compliance Report](https://docs.gitlab.com/ee/user/compliance/compliance_report/#separation-of-duties) is available that flags any MRs that did not properly go through two-person review or if the author or a committer to the MR self-approved their own code.

Reply:
Thank you for the correction, @samwhite-gl! I wasn't aware of this feature. Let me try to rephrase that bullet:

* *Verifying that two-party review happened will require more work to make it practical at scale*, and we don't want to block the build integrity piece of SLSA on this.

In particular::

* Most projects have some amount of historical unreviewed commits. How should SLSA handle this? We could require all these old commits to get reviewed, but that is impractical and of dubious value. We could require review of all old commits within the past X days to get reviewed, but that has the same problem on a smaller scale. We could require all new commits going forward to get reviewed, but then what if they turn it off again, or what if they turn it on right before building? I don't see an obvious best answer here.
* To scalably use the GitLab information, I think we'd need it in the form of a signed attestation rather than making the verifier call the API every time. So that piece would need to be built. That's also true of build provenance, but decoupling the two means that we can get build integrity done before tackling two-party review.